openmcyclus/depletion.py

```diff
@@ -3,156 +3,192 @@
 
-    def read_model(self):
+    def read_model(self, path):
         '''
         Read the .xml files in the defined path to create an OpenMC
         model in the python API. The files are assumed to be named
         geometry.xml, materials.xml, settings.xml, and (optionally)
         tallies.xml
 
+        Parameters:
+        -----------
+        path: str
+            path of directory holding the files for/from OpenMC
+
         Returns:
         ---------
         model: openmc.model.model object
             OpenMC model of reactor geometry.
         '''
-        model_kwargs = {"geometry": (self.path / "geometry.xml"),
-                        "materials": (self.path / "materials.xml"),
-                        "settings": (self.path / "settings.xml")}
-        tallies_path = self.path / "tallies.xml"
-        if tallies_path.exists():
-            model_kwargs["tallies"] = tallies_path
+        model_kwargs = {"geometry": path + "geometry.xml",
+                        "materials": path + "materials.xml",
+                        "settings": path + "settings.xml"}
         model = openmc.model.model.Model.from_xml(**model_kwargs)
         return model
 
-    def read_microxs(self):
+    def update_materials(self, comp_list, path):
+        '''
+        Read in the material compositions of the fuel assemblies present
+        in the reactor to be transmuted. Then modify the composition of
+        the pre-defined materials to match the compositions from
+        Cyclus.
+
+
+        Parameters:
+        -----------
+        comp_list: list of dicts
+            list of the fresh fuel compositions present in the core
+            at the calling of the transmute function.
+        path: str
+            path of directory holding the files for/from OpenMC
+
+        Returns:
+        --------
+        material_ids: list of strs
+            material id numbers for the OpenMC model
+
+        Outputs:
+        --------
+        materials.xml: file
+            updated XML for OpenMC with new compositions
+
+        '''
+        openmc_materials = ET.parse(str(path + "materials.xml"))
+        openmc_root = openmc_materials.getroot()
+        material_ids = []
+
+        for child in openmc_root:
+            if '_' in child.attrib['name']:
+                underscore_index = child.attrib['name'].index('_')
+                assembly_number = child.attrib['name'][underscore_index + 1:]
+                material_ids.append(child.attrib['id'])
+                if child.attrib['name'][:underscore_index] == 'assembly':
+                    for material in child.findall('nuclide'):
+                        child.remove(material)
+                    new_comp = comp_list[int(assembly_number) - 1]
+                    for nuclide in new_comp:
+                        Z = math.floor(nuclide / 10000000)
+                        A = math.floor((nuclide - Z * 10000000) / 10000)
+                        m = nuclide - Z * 10000000 - A * 10000
```

Comment:
I've made the same comment to Olek, scientific notation is superior
```suggestion
                        Z = math.floor(nuclide / int(1e7))
                        A = math.floor((nuclide - Z * int(1e7)) / int(1e5))
                        m = nuclide - Z * int(1e7)- A * int(1e5)
```
because it's more readable and more reliable. I immediately know the magnitude without having to count zeros, which is error-prone.